Update ClayBucketHandler.java

chavalitp opened the pull request:
Fill cauldron with clay water bucket

chavalitp committed:
Update ClayBucketHandler.java

Fill cauldron with clay water bucket

chavalitp committed:
Update ClayBucketHandler.java

Don't use water bucket if cauldron is full

TimeConqueror commented on src/main/java/iguanaman/iguanatweakstconstruct/claybuckets/ClayBucketHandler.java:
```diff
@@ -4,16 +4,41 @@
 
 public class ClayBucketHandler {
+    @SubscribeEvent
+    public void EntityInteract(PlayerInteractEvent event) {
```
Methods should start from lower case due to Java Conventions.

TimeConqueror replied:
One more thing: it probably should be called `onPlayerInteractWithBlock` or smth like this.

TimeConqueror commented on src/main/java/iguanaman/iguanatweakstconstruct/claybuckets/ClayBucketHandler.java:
```diff
@@ -4,16 +4,41 @@
+        ItemStack equipped = player.getCurrentEquippedItem();
+
+        if(block == null
+                || player == null
```
if the player was null, it would crash at line 23.

TimeConqueror commented on src/main/java/iguanaman/iguanatweakstconstruct/claybuckets/ClayBucketHandler.java:
```diff
@@ -4,16 +4,41 @@
+        EntityPlayer player = event.entityPlayer;
+        ItemStack equipped = player.getCurrentEquippedItem();
+
+        if(block == null
```
First of all block nullability is covered by `block instanceof BlockCauldron`

AFAIK: block can't be null

TimeConqueror commented on src/main/java/iguanaman/iguanatweakstconstruct/claybuckets/ClayBucketHandler.java:
```diff
@@ -4,16 +4,41 @@
+            return;
+        }
+
+        event.world.setBlockMetadataWithNotify(event.x, event.y, event.z, 3, 2);
```
use Blocks#cauldron#func_150024_a (stable mapping name: updateNeighborsAboutBlockChange) instead (because the same method is called when water bucket is used)

chavalitp committed:
Add RIGHT_CLICK_BLOCK check. Add creative mode check for bucket. Change setBlockMetadataWithNotify to setWaterLevel of BlockCauldron

Add RIGHT_CLICK_BLOCK check
Add creative mode check for bucket
Change setBlockMetadataWithNotify to setWaterLevel of BlockCauldron
